python/cuml/test/conftest.py

```diff
@@ -56,3 +61,54 @@ def housing_dataset():
     feature_names = data['feature_names']
 
     return X, y, feature_names
+
+
+def create_synthetic_dataset(generator=skl_make_reg,
+                             n_samples=100,
+                             n_features=10,
+                             test_size=0.25,
+                             random_state_generator=None,
+                             random_state_train_test_split=None,
+                             dtype=np.float32,
+                             **kwargs):
+    X, y = generator(
+        n_samples=n_samples,
+        n_features=n_features,
+        random_state=random_state_generator,
+        **kwargs
+    )
+
+    X_train, X_test, y_train, y_test = train_test_split(
+        X,
+        y,
+        test_size=test_size,
+        random_state=random_state_train_test_split
+    )
+
+    X_train = X_train.astype(dtype)
+    X_test = X_test.astype(dtype)
+    y_train = y_train.astype(dtype)
+    y_test = y_test.astype(dtype)
+
+    return X_train, X_test, y_train, y_test
+
+
+@pytest.fixture(scope="module")
+def exact_shap_regression_dataset():
+    return create_synthetic_dataset(generator=skl_make_reg,
+                                    n_samples=101,
+                                    n_features=11,
+                                    test_size=1,
+                                    random_state_generator=42,
+                                    random_state_train_test_split=42,
+                                    noise=0.1)
+
+
+@pytest.fixture(scope="module")
+def exact_shap_classification_dataset():
+    return create_synthetic_dataset(generator=skl_make_clas,
+                                    n_samples=101,
+                                    n_features=11,
+                                    test_size=1,
```

Comment:
@dantegd ping on this one? Maybe I'm just misunderstanding

Reply:
It was missing, just added a bigger test size to the exact tests of both perm and kernel (had to update the precalculated values because the seed changed when changing the sizes here)